### PR TITLE
SFTP: add settings for external traffic policy

### PR DESCRIPTION
**Pull request by stephaneetje**
Little PR to add support for externalTrafficPolicy configuration.

### CODE REVIEW

**Comment by stephaneetje** [comment]
@sj14 Ping

**Comment by sj14** [comment]
@stephaneetje please check the failed ci runs (run `make all`)

**Comment by stephaneetje** [comment]
> @stephaneetje please check the failed ci runs (run `make all`)

```
helm-charts @ stephaneetjemesian $ make all
cd tools && go install helm.sh/helm/v3/cmd/helm
go: downloading helm.sh/helm/v3 v3.14.3
go: downloading github.com/gofrs/flock v0.8.1
go: downloading github.com/Masterminds/semver/v3 v3.2.1
go: downloading github.com/gosuri/uitable v0.0.4
go: downloading github.com/moby/term v0.5.0
go: downloading github.com/spf13/cobra v1.8.0
go: downloading golang.org/x/term v0.17.0
go: downloading golang.org/x/text v0.14.0
go: downloading k8s.io/apimachinery v0.29.2
go: downloading k8s.io/client-go v0.29.2
go: downloading k8s.io/kubectl v0.29.2
go: downloading k8s.io/klog/v2 v2.120.1
go: downloading sigs.k8s.io/yaml v1.4.0
go: downloading github.com/fatih/color v1.16.0
go: downloading github.com/Masterminds/sprig/v3 v3.2.3
go: downloading k8s.io/api v0.29.2
go: downloading k8s.io/cli-runtime v0.29.2
go: downloading github.com/cyphar/filepath-securejoin v0.2.4
go: downloading github.com/mitchellh/copystructure v1.2.0
go: downloading k8s.io/apiextensions-apiserver v0.29.2
go: downloading github.com/xeipuuv/gojsonschema v1.2.0
go: downloading github.com/evanphx/json-patch v5.9.0+incompatible
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading github.com/Masterminds/vcs v1.13.3
go: downloading github.com/containerd/containerd v1.7.13
go: downloading oras.land/oras-go v1.2.5
go: downloading github.com/opencontainers/image-spec v1.1.0
go: downloading github.com/sirupsen/logrus v1.9.3
go: downloading github.com/Masterminds/squirrel v1.5.4
go: downloading github.com/jmoiron/sqlx v1.3.5
go: downloading github.com/rubenv/sql-migrate v1.6.1
go: downloading github.com/lib/pq v1.10.9
go: downloading golang.org/x/sys v0.17.0
go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.3
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading github.com/go-logr/logr v1.4.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/google/gofuzz v1.2.0
go: downloading github.com/mattn/go-runewidth v0.0.15
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.20
go: downloading github.com/BurntSushi/toml v1.3.2
go: downloading github.com/gobwas/glob v0.2.3
go: downloading golang.org/x/crypto v0.20.0
go: downloading github.com/Masterminds/goutils v1.1.1
go: downloading github.com/google/uuid v1.6.0
go: downloading github.com/huandu/xstrings v1.4.0
go: downloading github.com/shopspring/decimal v1.3.1
go: downloading github.com/imdario/mergo v0.3.16
go: downloading github.com/spf13/cast v1.6.0
go: downloading k8s.io/utils v0.0.0-20240102154912-e7106e64919e
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.4.1
go: downloading github.com/fvbommel/sortorder v1.1.0
go: downloading github.com/liggitt/tabwriter v0.0.0-20181228230101-89fcab3d43de
go: downloading github.com/google/gnostic-models v0.6.8
go: downloading golang.org/x/sync v0.6.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading k8s.io/kube-openapi v0.0.0-20240224005224-582cce78233b
go: downloading sigs.k8s.io/kustomize/api v0.16.0
go: downloading sigs.k8s.io/kustomize/kyaml v0.16.0
go: downloading golang.org/x/net v0.21.0
go: downloading github.com/golang/protobuf v1.5.3
go: downloading github.com/hashicorp/errwrap v1.1.0
go: downloading github.com/mitchellh/reflectwalk v1.0.2
go: downloading github.com/xeipuuv/gojsonreference v0.0.0-20180127040603-bd5ef7bd5415
go: downloading github.com/google/go-cmp v0.6.0
go: downloading k8s.io/component-base v0.29.2
go: downloading github.com/exponent-io/jsonpath v0.0.0-20210407135951-1de76d718b3f
go: downloading github.com/opencontainers/go-digest v1.0.0
go: downloading github.com/docker/cli v25.0.3+incompatible
go: downloading github.com/docker/distribution v2.8.3+incompatible
go: downloading github.com/docker/docker v25.0.5+incompatible
go: downloading github.com/docker/go-connections v0.5.0
go: downloading github.com/russross/blackfriday/v2 v2.1.0
go: downloading github.com/lann/builder v0.0.0-20180802200727-47ae307949d0
go: downloading github.com/go-gorp/gorp/v3 v3.1.0
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/asaskevich/govalidator v0.0.0-20230301143203-a9d515a09cc2
go: downloading k8s.io/apiserver v0.29.2
go: downloading github.com/rivo/uniseg v0.4.7
go: downloading sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd
go: downloading github.com/chai2010/gettext-go v1.0.2
go: downloading github.com/MakeNowJust/heredoc v1.0.0
go: downloading github.com/mitchellh/go-wordwrap v1.0.1
go: downloading github.com/gregjones/httpcache v0.0.0-20190611155906-901d90724c79
go: downloading github.com/peterbourgon/diskv v2.0.1+incompatible
go: downloading github.com/json-iterator/go v1.1.12
go: downloading google.golang.org/protobuf v1.33.0
go: downloading golang.org/x/oauth2 v0.17.0
go: downloading golang.org/x/time v0.5.0
go: downloading github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb
go: downloading github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
go: downloading github.com/moby/locker v1.0.1
go: downloading github.com/go-openapi/swag v0.22.9
go: downloading github.com/go-openapi/jsonreference v0.20.4
go: downloading google.golang.org/grpc v1.62.0
go: downloading github.com/containerd/log v0.1.0
go: downloading github.com/docker/docker-credential-helpers v0.8.1
go: downloading github.com/lann/ps v0.0.0-20150810152359-62de8c46ede0
go: downloading github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
go: downloading github.com/google/btree v1.1.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading gopkg.in/evanphx/json-patch.v5 v5.9.0
go: downloading github.com/go-errors/errors v1.5.1
go: downloading go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.49.0
go: downloading go.opentelemetry.io/otel v1.24.0
go: downloading go.opentelemetry.io/otel/trace v1.24.0
go: downloading github.com/distribution/reference v0.5.0
go: downloading github.com/klauspost/compress v1.17.7
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/go-openapi/jsonpointer v0.20.2
go: downloading github.com/gorilla/websocket v1.5.1
go: downloading github.com/emicklei/go-restful/v3 v3.11.3
go: downloading github.com/monochromegane/go-gitignore v0.0.0-20200626010858-205db1a8cc00
go: downloading github.com/xlab/treeprint v1.2.0
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/felixge/httpsnoop v1.0.4
go: downloading go.opentelemetry.io/otel/metric v1.24.0
go: downloading github.com/gorilla/mux v1.8.1
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/docker/go-metrics v0.0.1
go: downloading github.com/go-logr/stdr v1.2.2
go: downloading github.com/moby/spdystream v0.2.0
go: downloading go.starlark.net v0.0.0-20240123142251-f86470692795
go: downloading github.com/mxk/go-flowrate v0.0.0-20140419014527-cca7078d478f
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20240221002015-b0ce06bbee7c
go: downloading github.com/prometheus/client_golang v1.18.0
go: downloading github.com/prometheus/common v0.48.0
go: downloading github.com/cespare/xxhash/v2 v2.2.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/prometheus/client_model v0.6.0
go: downloading github.com/prometheus/procfs v0.12.0
./scripts/test.sh
./bin/helm custom.yaml
./bin/helm empty.yaml
./bin/helm extraManifests.yaml
./bin/helm nodePort.yaml
./bin/helm persistent.yaml
cd tools && go install github.com/norwoodj/helm-docs/cmd/helm-docs
go: downloading github.com/norwoodj/helm-docs v1.13.1
go: downloading github.com/spf13/viper v1.18.2
go: downloading k8s.io/helm v2.17.0+incompatible
go: downloading github.com/spf13/afero v1.11.0
go: downloading github.com/sagikazarmark/slog-shim v0.1.0
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/fsnotify/fsnotify v1.7.0
go: downloading github.com/mitchellh/mapstructure v1.5.0
go: downloading github.com/magiconair/properties v1.8.7
go: downloading github.com/subosito/gotenv v1.6.0
go: downloading github.com/pelletier/go-toml/v2 v2.1.1
./bin/helm-docs
INFO[2024-03-26T19:15:51+01:00] Found Chart directories [charts/sftp-server]
INFO[2024-03-26T19:15:51+01:00] Generating README Documentation for chart charts/sftp-server
```

I don't see the errors there

**Comment by sj14** [comment]
No errors are expected. It regenerates files which need to be committed.

File: charts/sftp-server/templates/service.yaml [hunk 1]
```diff
@@ -6,6 +6,11 @@ metadata:
     {{- include "sftp-server.labels" . | nindent 4 }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if $.Values.service.externalTraffic.enabled }}
+  {{- if eq "ClusterIP" $.Values.service.serviceType }}
```
Comment (sj14) [inline charts/sftp-server/templates/service.yaml:10]: Is the `externalTrafficPolicy` limited to the `ClusterIP` service? I didn't see this.

https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip

File: charts/sftp-server/values.yaml [hunk 2]
```diff
@@ -117,6 +117,9 @@ service:
   port: 22
   # -- When using the NodePort type, you can specify a fixed port ([Kubernetes docs](https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport))
   nodePort: 0 
+  externalTraffic:
+    enabled: true
```
Comment (sj14) [inline charts/sftp-server/values.yaml:121]: I suggest to disable it by default, as that's also the k8s default.

```suggestion
    enabled: false
```

File: charts/sftp-server/templates/service.yaml (hunk 1 above)
Comment (stephaneetje) [reply charts/sftp-server/templates/service.yaml]: Indeed you're right, i deleted that if

File: charts/sftp-server/values.yaml (hunk 2 above)
Comment (stephaneetje) [reply charts/sftp-server/values.yaml]: Fixed

**Comment by sj14** [comment]
Thanks a lot for your contribution 🎉

**Pull request merged**